http/negotiation.ts

```diff
@@ -12,6 +12,13 @@ import { preferredEncodings } from "./_negotiation/encoding.ts";
 import { preferredLanguages } from "./_negotiation/language.ts";
 import { preferredMediaTypes } from "./_negotiation/media_type.ts";
 
+/**
+ * The part of the {@linkcode Request} interface that the methods requires,
+ * which allows for the working with objects which may not be strictly `Request`
+ * objects.
+ */
+export type Headered = { headers: { get(name: string): string | null } };
```

Comment:
@kitsonk 
A question about some details. Can this be `{ headers: Headers }` in your use case?

Reply:
Yes, I'd prefer this be `Headers` too.

Reply:
It feels a bit undesirable to expose this type as a public API. If it's ok to be `{ headers: Headers }`, maybe we can use `Pick<Request, "headers">` type instead of `Headered` for function parameter types